en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,109 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some block will require environment parameters.
+Those parameters could be of multiple type, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when we call a block.
+
+This easy solution is not optimal a main reason, when working with `ESI`_, you will cache each block
+for each different url.
+
+``/block/1`` will be in one cache entry.
+``/block/1?foo=bar`` will be in another cache entry.
```

Comment:
You should precise why it is not optimal: if foo=bar is not usefull for the block rendering, this info makes two entries for the same html